#!/usr/bin/env python

PACKAGE = "rm_digitalimg_proc_deep"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("score_threshold", double_t, 0, "Description of score threshold", 0.3, 0.01, 0.99)
gen.add("nms_threshold", double_t, 0, "Description of nms threshold", 0.3, 0.01, 0.99)

target_color = gen.enum([ gen.const("BLUE", int_t, 0, "BLUE"),
                          gen.const("RED", int_t, 1, "RED")], "Color filtering options")

gen.add("target_color", int_t, 0, "Color filtering", 0, 0, 1, edit_method=target_color)

draw_type = gen.enum([ gen.const("DISABLE", int_t, 0, "DISABLE"),
                       gen.const("RAW", int_t, 1, "RAW"),
                       gen.const("BINARY", int_t, 2, "BINARY"),
                       gen.const("MORPHOLOGY", int_t, 3, "MORPHOLOGY"),
                       gen.const("BARS", int_t, 4, "BARS"),
                       gen.const("ARMOR", int_t, 5, "ARMOR"),
                       gen.const("TRACK", int_t, 6, "TRACK")], "Drawing options")

gen.add("draw_type", int_t, 0, "Drawing options", 0, 0, 6, edit_method=draw_type)

exit(gen.generate(PACKAGE, "rm_digitalimg_proc_deep", "Inference"))